youtube_dl/extractor/duboku.py

```diff
@@ -0,0 +1,242 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import re
+
+from .common import InfoExtractor
+from ..compat import compat_urlparse
+from ..utils import (
+    clean_html,
+    extract_attributes,
+    ExtractorError,
+    get_elements_by_class,
+    int_or_none,
+    js_to_json,
+    smuggle_url,
+    unescapeHTML,
+)
+
+
+def _get_elements_by_tag_and_attrib(html, tag=None, attribute=None, value=None, escape_value=True):
+    """Return the content of the tag with the specified attribute in the passed HTML document"""
+
+    if tag is None:
+        tag = '[a-zA-Z0-9:._-]+'
+    if attribute is None:
+        attribute = ''
+    else:
+        attribute = r'\s+(?P<attribute>%s)' % re.escape(attribute)
+    if value is None:
+        value = ''
+    else:
+        value = re.escape(value) if escape_value else value
+        value = '=[\'"]?(?P<value>%s)[\'"]?' % value
+
+    retlist = []
+    for m in re.finditer(r'''(?xs)
+        <(?P<tag>%s)
+         (?:\s+[a-zA-Z0-9:._-]+(?:=[a-zA-Z0-9:._-]*|="[^"]*"|='[^']*'|))*?
+         %s%s
+         (?:\s+[a-zA-Z0-9:._-]+(?:=[a-zA-Z0-9:._-]*|="[^"]*"|='[^']*'|))*?
+        \s*>
+        (?P<content>.*?)
+        </\1>
+    ''' % (tag, attribute, value), html):
+        retlist.append(m)
+
+    return retlist
+
+
+def _get_element_by_tag_and_attrib(html, tag=None, attribute=None, value=None, escape_value=True):
+    retval = _get_elements_by_tag_and_attrib(html, tag, attribute, value, escape_value)
+    return retval[0] if retval else None
+
+
+class DubokuIE(InfoExtractor):
+    IE_NAME = 'duboku'
+    IE_DESC = 'www.duboku.co'
+
+    _VALID_URL = r'(?:https?://[^/]+\.duboku\.co/vodplay/)(?P<id>[0-9]+-[0-9-]+)\.html.*'
```

Comment:
Require n-n-n in `id` field; no need to match the tail:
```suggestion
    _VALID_URL = r'(?:https?://[^/]+\.duboku\.co/vodplay/)(?P<id>(?:[0-9]+-){2}[0-9]+)\.html'
```